projects/packages/my-jetpack/src/products/class-search-stats.php

```diff
@@ -36,6 +36,8 @@ class Search_Stats {
 	const CACHE_EXPIRY                  = 1 * MINUTE_IN_SECONDS;
 	const CACHE_GROUP                   = 'jetpack_search';
 	const POST_TYPE_BREAKDOWN_CACHE_KEY = 'post_type_break_down';
+	const TOTAL_POSTS_COUNT_CACHE_KEY   = 'total-post-count';
+	const POST_COUNT_QUERY_LIMIT        = 1e6;
```

Comment:
My only question is if we should lower this limit to something closer to 100k to make it less likely to cause even minor performance problems on slower hosting.

Reply:
@gibrown this sounds reasonable to me and is actually the query limit we used to fix the other similar issue. I did some testing to try to determine where things really start to break down in extreme cases but I think you're right and erring on the side of caution always makes sense so I've updated this to 100k.